tests/validation/cattlevalidationtest/core/test_services.py

```diff
@@ -0,0 +1,920 @@
+
+    service1, env = create_env_and_svc(client, launch_config,
+                                       scale)
+
+    service2 = create_svc(client, env, launch_config, scale)
+
+    # Environment Activate Services
+    env = env.activateservices()
+
+    service1 = client.wait_success(service1, 300)
+    assert service1.state == "active"
+    check_container_in_service(super_client, service1)
+
+    service2 = client.wait_success(service2, 300)
+    assert service2.state == "active"
+    check_container_in_service(super_client, service2)
+
+    # Environment Deactivate Services
+    env = env.deactivateservices()
+
+    time.sleep(60)
+    service1 = client.wait_success(service1, 300)
+    assert service1.state == "inactive"
+    check_stopped_container_in_service(super_client, service1)
+
+    service2 = client.wait_success(service2, 300)
+    assert service2.state == "inactive"
+    check_stopped_container_in_service(super_client, service2)
+
+    # Environment Activate Services
+    env = env.activateservices()
+
+    service1 = client.wait_success(service1, 300)
+    assert service1.state == "active"
+    check_container_in_service(super_client, service1)
+
+    service2 = client.wait_success(service2, 300)
+    assert service2.state == "active"
+    check_container_in_service(super_client, service2)
+
+    # Delete Environment
+    env = client.wait_success(client.delete(env))
+    assert env.state == "removed"
+
+    # Deleting service results in instances of the service to be "removed".
+    # instance continues to be part of service , until the instance is purged.
+
+    check_for_deleted_service(super_client, env, service1)
+    check_for_deleted_service(super_client, env, service2)
+
+    delete_all(client, [env])
+
+
+def test_service_activate_deactivate_delete(super_client, client,
+                                            socat_containers):
+
+    launch_config = {"imageUuid": TEST_IMAGE_UUID}
+
+    scale = 2
+
+    service, env = create_env_and_svc(client, launch_config,
+                                      scale)
+    # Activate Services
+    service = service.activate()
+    service = client.wait_success(service, 300)
+    assert service.state == "active"
+
+    check_container_in_service(super_client, service)
+
+    # Deactivate Services
+    service = service.deactivate()
+    service = client.wait_success(service, 300)
+    assert service.state == "inactive"
+    time.sleep(60)
+
+    check_stopped_container_in_service(super_client, service)
+
+    # Activate Services
+    service = service.activate()
+    service = client.wait_success(service, 300)
+    assert service.state == "active"
+
+    check_container_in_service(super_client, service)
+
+    # Delete Service
+    service = client.wait_success(client.delete(service))
+    assert service.state == "removed"
+
+    check_for_deleted_service(super_client, env, service)
+
+    delete_all(client, [env])
+
+
+def test_service_activate_stop_instance_deactivate_activate(
```

Comment:
This might be a redundant test cases since deactivating a service means stopping containers already.